lib/matplotlib/lines.py

```diff
@@ -790,8 +790,21 @@ def draw(self, renderer):
                     cap = self._dashcapstyle
                     join = self._dashjoinstyle
                 else:
+                    if tpath.should_simplify and affine.has_inverse:
+                        # the path simplification code expects the path
+                        # to be in pixel space, not data space, but the
+                        # renderer downstream expects the path to be
+                        # in data space. The transformation `affine`
+                        # will put the path into pixel space, and
+                        # affine.inverted() will transform it back to
+                        # data space. So, transform using affine, simplify
+                        # the path, and transform using the affine inverted.
+                        tpath = affine.inverted().transform_path(
+                            affine.transform_path(tpath).cleaned(simplify=True)
+                        )
```

Comment:
Sorry didn't see your 2nd message before posting. So I guess it looks like `affine` is being used for something else later on, I'm guessing in the clipping routine:

```<path_converters.h>
/*
 This file contains a number of vertex converters that modify
 paths. They all work as iterators, where the output is generated
 on-the-fly, and don't require a copy of the full data.

 Each class represents a discrete step in a "path-cleansing" pipeline.
 They are currently applied in the following order in the Agg backend:

   ...

   3. PathClipper: Clips line segments to a given rectangle.  This is
      helpful for data reduction, and also to avoid a limitation in
      Agg where coordinates can not be larger than 24-bit signed
      integers.

   ...

   5. PathSimplifier: Removes line segments from highly dense paths
      that would not have an impact on their appearance.  Speeds up
      rendering and reduces file sizes.

   ...
 */
```